src/adapters/fetch.js

```diff
@@ -23,8 +23,11 @@ export default function (fetch) {
     toJSON(resp).then((data)=> {
       if (resp.status >= 200 && resp.status < 300) {
         return data;
+      } else if (resp.status >= 400) {
```

Comment:
I think that the best way to refactor code.
```javascript
if (resp.status >= 400) { 
   return Promise.reject({ status: resp.status, statusText: resp.statusText });
} else {
   return toJSON(resp).then(.......)
}
``

Reply:
👍  Makes sense what you propose. My mindset was on group all the statusCode